SamplePay/app/src/main/java/com/example/android/samplepay/ui/MainActivity.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.samplepay.ui
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.activity.enableEdgeToEdge
+import androidx.activity.viewModels
+import androidx.compose.runtime.getValue
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+
+class MainActivity : ComponentActivity() {
+
+    private val viewModel: PaymentViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        enableEdgeToEdge()
+        super.onCreate(savedInstanceState)
+
+        // Refrain from taking action if the intent data is inappropriate
+        if ("org.chromium.intent.action.PAY" == intent.action && intent.extras == null) {
+            cancel()
+        }
+
+        // TODO CHECK IF THIS AUTHORIZATION STILL NEEDS TO BE MADE (consider doing in activity)
+        /*
+        _error.value = if (application.authorizeCaller(callingPackage)) {
```

Comment:
@rsolomakhin is this check necessary now that non-Chrome callers are also allowed?

Reply:
Alright, how about we persist/update the package name and fingerprint of the caller every time a payment process is started (eg.: calling PAY) and verify subsequent update payment details calls against these values?

If we do that, we probably want to cancel the whole payment process and prompt the user to start again if there are inconsistencies. For example, suppose that you have two websites open that interact with the payment app, and, by mistake, fiddle with checkout values between pages. Is that even possible though?

Reply:
If a user launches the payment app from browser A and then switched to browser B, then it would be possible for the app to receive a secondary `PAY` intent.

If that happens and the payment app does not support it, then it would be best for the second `PAY` intent to show a message to the user that the payment flow is already running elsewhere.

Some payment apps may be able to handle multiple `PAY` intents at the same time, but I don't think that a demo app should be worrying about that.